autometa/common/external/diamond.py

```diff
@@ -182,50 +337,60 @@ def blast(fasta, database, outfpath, blast_type='blastp', evalue=float('1e-5'),
         empty = not os.stat(outfpath).st_size
         if not empty:
             if verbose:
-                logger.warning(f'FileExistsError: {outfpath}. To overwrite use --force')
+                logger.warning(f"FileExistsError: {outfpath}. To overwrite use --force")
             return outfpath
     blast_type = blast_type.lower()
-    if blast_type not in ['blastp','blastx']:
-        raise ValueError(f'blast_type must be blastp or blastx. Given: {blast_type}')
+    if blast_type not in ["blastp", "blastx"]:
+        raise ValueError(f"blast_type must be blastp or blastx. Given: {blast_type}")
     if verbose:
-        logger.debug(f'Diamond{blast_type.title()} {fasta} against {database}')
+        logger.debug(f"Diamond{blast_type.title()} {fasta} against {database}")
     cmd = [
-        'diamond',
-        'blastp',
-        '--query',
+        "diamond",
+        blast_type,
+        "--query",
         fasta,
-        '--db',
+        "--db",
         database,
-        '--evalue',
+        "--evalue",
         evalue,
-        '--max-target-seqs',
+        "--max-target-seqs",
         maxtargetseqs,
-        '--threads',
-        cpus,
-        '--outfmt',
-        '6',
-        '--out',
+        "--threads",
+        nproc,
+        "--outfmt",
+        "6",
+        "--out",
         outfpath,
-        '--tmpdir',
-        tmpdir]
+        "--tmpdir",
+        tmpdir,
+    ]
     cmd = [str(c) for c in cmd]
     if verbose:
         logger.debug(f'RunningDiamond: {" ".join(cmd)}')
-    with open(os.devnull, 'w') as stdout, open(os.devnull, 'w') as stderr:
-        proc = subprocess.run(cmd, stdout=stdout, stderr=stderr)
-    if proc.returncode:
-        raise OSError(f'DiamondFailed:\nArgs:{proc.args}\nReturnCode:{proc.returncode}')
+    proc = subprocess.run(
+        cmd, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL, shell=True, text=True
+    )
+    try:
+        proc.check_returncode()
+    except subprocess.CalledProcessError as err:
+        raise err
     return outfpath
 
-def parse(results, top_pct=0.9, verbose=False):
-    """Retrieve diamond results from output table
+
+def parse(results, bitscr_thresh=0.9, verbose=False):
```

Comment:
I'm not sure `bitscr_thresh` is clear as a variable name. When inspecting blast results, the bitscore is usually a float that can range to the 1000s. However, we are asking for a float between 0 and 1. I think this is confusing.